programs/marinade-finance/src/stake_system/deposit_stake_account.rs

```diff
@@ -28,9 +29,7 @@ impl<'info> DepositStakeAccount<'info> {
             .check_msol_mint(self.msol_mint.to_account_info().key)?;
         self.state
             .check_msol_mint_authority(self.msol_mint_authority.key)?;
-        check_owner_program(self.stake_account.as_ref(), &stake::program::ID, "stake")?;
         check_token_mint(&self.mint_to, self.state.msol_mint, "mint_to")?;
-        check_owner_program(&self.rent_payer, &system_program::ID, "rent_payer")?;
```

Comment:
This is not duplicated but it is OK to remove it anyways because there is no real reason to check this here